engine/apps/api/permissions/__init__.py

```diff
@@ -1,5 +1,200 @@
+        ALERT_RECEIVE_CHANNELS_WRITE = _generate_permission_string(Resources.ALERT_RECEIVE_CHANNELS, Actions.WRITE)
+
+        INTEGRATIONS_READ = _generate_permission_string(Resources.INTEGRATIONS, Actions.READ)
+        INTEGRATIONS_WRITE = _generate_permission_string(Resources.INTEGRATIONS, Actions.WRITE)
+
+        ESCALATION_CHAINS_READ = _generate_permission_string(Resources.ESCALATION_CHAINS, Actions.READ)
+        ESCALATION_CHAINS_WRITE = _generate_permission_string(Resources.ESCALATION_CHAINS, Actions.WRITE)
+
+        SCHEDULES_READ = _generate_permission_string(Resources.SCHEDULES, Actions.READ)
+        SCHEDULES_WRITE = _generate_permission_string(Resources.SCHEDULES, Actions.WRITE)
+
+        CHATOPS_READ = _generate_permission_string(Resources.CHATOPS, Actions.READ)
+        CHATOPS_WRITE = _generate_permission_string(Resources.CHATOPS, Actions.WRITE)
+
+        OUTGOING_WEBHOOKS_READ = _generate_permission_string(Resources.OUTGOING_WEBHOOKS, Actions.READ)
+        OUTGOING_WEBHOOKS_WRITE = _generate_permission_string(Resources.OUTGOING_WEBHOOKS, Actions.WRITE)
+
+        MAINTENANCE_READ = _generate_permission_string(Resources.MAINTENANCE, Actions.READ)
+        MAINTENANCE_WRITE = _generate_permission_string(Resources.MAINTENANCE, Actions.WRITE)
+
+        API_KEYS_READ = _generate_permission_string(Resources.API_KEYS, Actions.READ)
+        API_KEYS_WRITE = _generate_permission_string(Resources.API_KEYS, Actions.WRITE)
+
+        ONCALL_SHIFTS_READ = _generate_permission_string(Resources.ONCALL_SHIFTS, Actions.READ)
+        ONCALL_SHIFTS_WRITE = _generate_permission_string(Resources.ONCALL_SHIFTS, Actions.WRITE)
+
+        NOTIFICATION_SETTINGS_READ = _generate_permission_string(Resources.NOTIFICATION_SETTINGS, Actions.READ)
+        NOTIFICATION_SETTINGS_WRITE = _generate_permission_string(Resources.NOTIFICATION_SETTINGS, Actions.WRITE)
+
+        GLOBAL_SETTINGS_READ = _generate_permission_string(Resources.GLOBAL_SETTINGS, Actions.READ)
+        GLOBAL_SETTINGS_WRITE = _generate_permission_string(Resources.GLOBAL_SETTINGS, Actions.WRITE)
+
+        OWN_SETTINGS_READ = _generate_permission_string(Resources.OWN_SETTINGS, Actions.READ)
+        OWN_SETTINGS_WRITE = _generate_permission_string(Resources.OWN_SETTINGS, Actions.WRITE)
+
+        OTHERS_SETTINGS_READ = _generate_permission_string(Resources.OTHERS_SETTINGS, Actions.READ)
+        OTHERS_SETTINGS_WRITE = _generate_permission_string(Resources.OTHERS_SETTINGS, Actions.WRITE)
+
+        ORGANIZATIONS_READ = _generate_permission_string(Resources.ORGANIZATIONS, Actions.READ, True)
+        TEAMS_READ = _generate_permission_string(Resources.TEAMS, Actions.READ, True)
+        USERS_READ = _generate_permission_string(Resources.USERS, Actions.READ, True)
+
+        TESTING = _generate_permission_string(Resources._IMPOSSIBLE_TESTING, Actions._IMPOSSIBLE_TESTING)
+
+    ViewSetOrAPIView = typing.Union[ViewSet, APIView]
+
+    @staticmethod
+    def user_has_permissions(
```

Comment:
@gamab tagging you for visibility. This is bulk of how we will handle authorization logic in the OnCall backend.